storage: add trace point in GCQueue

tbg opened the pull request:
We really want these added bits of information to show up in the
traces (and not only in the queue event log).

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/9394)

<!-- Reviewable:end -->

tamird commented:
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[storage/gc_queue.go, line 318 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394#-KRijML6WPQ-qSLy_ojU:-KRijML6WPQ-qSLy_ojV:b-5hlkjh) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4a3f1cf4ccfa52350b6706334cf1d1daa3a4103/storage/gc_queue.go#L318)):*
> ```Go
> 
> 	log.Infof(gcq.ctx, "completed with stats %+v", info)
> 	log.Tracef(ctx, "completed with stats %+v", info)
> ```

heh, why are there two different contexts here? cc @raduberinde 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/gc_queue.go, line 318 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394#-KRijML6WPQ-qSLy_ojU:-KRilnk8U8NNLEtF9xKQ:b1cisgf) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4a3f1cf4ccfa52350b6706334cf1d1daa3a4103/storage/gc_queue.go#L318)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> heh, why are there two different contexts here? cc @raduberinde 
</details>
Deserves a comment - one goes to the queue's context (which ends up in an event log). The other goes into that operation's span.

---

*[storage/gc_queue.go, line 336 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394#-KRimIcXevYWfKB7cWQX:-KRimIcXevYWfKB7cWQY:b-hm4vee) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4a3f1cf4ccfa52350b6706334cf1d1daa3a4103/storage/gc_queue.go#L336)):*
> ```Go
> 	ba.Add(&gcArgs)
> 	if _, pErr := repl.Send(ctx, ba); pErr != nil {
> 		log.Tracef(ctx, "error: %+v", pErr)
> ```

`ErrEventf`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394)*
<!-- Sent from Reviewable.io -->

tbg committed:
storage: add trace point in GCQueue

We really want these added bits of information to show up in the
traces (and not only in the queue event log).

tschottdorf commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[storage/gc_queue.go, line 318 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394#-KRijML6WPQ-qSLy_ojU:-KRj-rXX7nbFzDKBcmUe:b-gm0dj3) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4a3f1cf4ccfa52350b6706334cf1d1daa3a4103/storage/gc_queue.go#L318)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> Deserves a comment - one goes to the queue's context (which ends up in an event log). The other goes into that operation's span.
</details>
Done, but I think I'm of the (unreflected) opinion that we shouldn't ever have to think about there being two contexts. It would be easier if what is `ctx` now also contains the EventLog and operations simply hit both. That way `gq.ctx` can be used everywhere, and in places which have "foreground work" we'd derive a context for it. Or something like that.

---

*[storage/gc_queue.go, line 336 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394#-KRimIcXevYWfKB7cWQX:-KRj-r6cDVtEgCfoRG1I:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4a3f1cf4ccfa52350b6706334cf1d1daa3a4103/storage/gc_queue.go#L336)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> `ErrEventf`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/gc_queue.go, line 318 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394#-KRijML6WPQ-qSLy_ojU:-KRj0jWnYWXXLE344F8l:b-ypnp7d) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4a3f1cf4ccfa52350b6706334cf1d1daa3a4103/storage/gc_queue.go#L318)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> Done, but I think I'm of the (unreflected) opinion that we shouldn't ever have to think about there being two contexts. It would be easier if what is `ctx` now also contains the EventLog and operations simply hit both. That way `gq.ctx` can be used everywhere, and in places which have "foreground work" we'd derive a context for it. Or something like that.
</details>
:+1: 

In general, there are probably too many tracing/event/logging functions and they're ripe for a culling.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394#-:-KRj0rUxE7rlt0cFSE9G:bnfp4nl)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
*[storage/gc_queue.go, line 318 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394#-KRijML6WPQ-qSLy_ojU:-KRj0qIUWYiyqq7HSqiK:baslphp) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4a3f1cf4ccfa52350b6706334cf1d1daa3a4103/storage/gc_queue.go#L318)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> :+1: 
> 
> In general, there are probably too many tracing/event/logging functions and they're ripe for a culling.
</details>
@tschottdorf We can do something like that (attach the queue's event log context as well). The problem (as a general solution) is that you normally don't want *all* traces for `ctx` to pollute the queue's event log..

@tamird: Tracef functions are deprecated in favor of Event, just didn't fix the callsites because of the branching stuff.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9394)*
<!-- Sent from Reviewable.io -->
